openms-wiki/developer-faq.md

```diff
@@ -0,0 +1,490 @@
+* Read the OpenMS Coding Convention.
+* Read the OpenMS Tutorial.
+* Create a GitHub account
+* Register to the open-ms-general and open-ms-developers mailing list. (You can see the developers list only if you are logged in to Sourceforge and if you are a OpenMS developer).
+
+### What is the difference between an OpenMS tool and util?
+
+A tool starts its lifecycle in `UTILS` and may exist without being thoroughly tested. Tools may be promoted from `UTILS` to `TOOLS` if they are stable enough, are fully tested, fully documented, and a test workflow exists.
+
+### I have written a class for OpenMS. What should I do?
+
+Follow the [OpenMS coding conventions](coding-conventions.md).
+
+Coding style (brackets, variable names, etc.) must conform to the conventions.
+
+* The class and all the members must be documented thoroughly.
+* You can check your code with the tool  `tools/checker.php`. Call `php tools/checker.php` for detailed instructions.
+
+Please open a pull request and follow the [checklist](pull-request-checklist.md).
+
+### How do I update the [openms.de](https://www.openms.de)?
+
+Login to the [wordpress admin area](www.openms.de/wp-admin) with your username and password assigned by the current Homepage maintainers.
+
+### Can I use QT designer to create GUI widgets?
+
+Yes. If you want to create a class called `Widget: Create .ui-File` with `QT designer` and store it as `Widget.ui.`, add the class to  `sources.cmake`.
+From the .ui-File the file `include/OpenMS/VISUAL/UIC/ClassTemplate.h` is generated by the build system.
+DO NOT CHECK-IN THIS FILE, AS IT IS GENERATED AUTOMATICALLY, WHEN NEEDED!!!
+Derive the class `Widget` from `WidgetTemplate`. You need to check in the `Widget.h` and `Widget.cpp` files.
+
+### Can the START_SECTION-macro not handle template methods that have two or more arguments?
+
+Put round brackets around the method declaration. Then it should work.
+
+### Where can I find the binary installers created?
+
+View the binary installers at the [build archive](https://abibuilder.informatik.uni-tuebingen.de/archive/openms/OpenMSInstaller/nightly/).
+Please verify the creation date of the individual installers, as there may have been an error while creating the installer.
+
+## Troubleshooting
+
+The following section provides information about how to troubleshoot common OpenMS issues.
+
+### OpenMS complains about boost not being found but I'm sure its there!
+
+`CMake` got confused. Set up a new build directory and try again. If you build in-source (not recommended), deleting the `CMakeCache.txt` and `cmake` directory might help.
+
+## Build System
+
+The following questions are related to the build system.
+
+### What is CMake?
+
+`CMake` builds BuildSystems for different platforms, e.g. VisualStudio Solutions on Windows, Makefiles on Linux etc.
+This allows us to define in one central location (namely `CMakeLists.txt`) how OpenMS is build and have the platform specific stuff handled by `CMake`.
```

Comment:
Delete:

> us.

Reply:
I don't think that works. How about:
`
Therefore, one central location can be used to define how OpenMS is built and include the platform specific instructions handled by `CMake`.`